draft-dekater-scion-pki.md

```diff
@@ -1303,21 +1327,57 @@ The steps required to create a new AS certificate are the following:
 
 # Security Considerations
 
-The entire document is about security considerations.
-More details will follow in future versions of this draft.
+The goal of SCION is to provide a secure inter-domain network architecture, therefore this paragraph focuses on *inter*-AS security considerations. Specifically, it covers potential attacks and mitigations for the control-plane PKI. All *intra*-AS trust- and security aspects are out of scope.
+
+
+## Dependency on Certificates
+
+In public key infrastructures, certificate authorities have both the responsibility and power to issue and revoke certificates. If a CA is compromised or malicious, this power can be abused. A misbehaving CA could refuse to issue certificates to legitimate entities, or even issue illegitimate certificates to allow impersonation of another entity. In the context of the SCION control plane PKI, refusing to issue or renew a certificate to an AS will ultimately cut that AS off from the network, turning the CP-PKI into a potential network "kill switch". In order to prevent this, within each ISD there are usually multiple independent CAs.
+
+SCION’s trust architecture fundamentally differs from a global monopolistic trust model. In SCION, each ISD manages its own trust roots instead of a single global entity providing those roots. This structure gives each ISD autonomy in terms of key management and in terms of trust, and prevents the occurrence of a global kill switch affecting all ISDs at once. However, each ISD is still susceptible of compromises that could affect or halt other components (control plane and forwarding). The following sections explain these cases and possible countermeasures.
+
+
+### Compromise of an ISD
+
+Compared to DNSSEC and RPKI, in SCION there is no central authority that could "switch off" an ISD, as each ISD relies on its own independent cryptographic trust roots. Each AS within an ISD is therefore dependant on its ISD's PKI for its functioning. This section discusses potential compromises of the PKI at various levels of the hierarchy.
+
+- On TRC level: The private root keys of the root certificates contained in an TRC are used to sign CA certificates. If one of these private root keys is compromised, the adversary could issue illegitimate CA certificates which may be used in further attacks. To maliciously  perform a TRC update, an attacker would need to compromise multiple voting keys. This number depends on the voting quorum set in the TRC: the higher the quorum, the more unlikely a malicious update will be.
+- On CA level: The private keys of an ISD's CA certificates are used to sign the AS certificates. All ASes within an ISD obtain certificates directly from the CAs. If one of the CA’s keys is compromised, an adversary could issue illegitimate AS certificates, which may be used in further attacks.
```

Comment:
@matzf so there is no revocation for CA certificates?  If one of it is compromised, do you just have to wait that it rolls over (typically 11 days)? Shall we add something more on recovery from compromise?

Reply:
Correct, there is no revocation, only expiration or TRC updates.
For a CA compromise, the alternative to waiting for expiration is to change the root key by updating the TRC. There is still a large-ish window for attacks during the time it takes for the TRC to be distributed and the grace period. This should be comparably shorter than expiration (I'd assume something on the order of dozens of minutes to few hours at most).

To me, only discussing (the lack of) revocations in the security considerations section does indeed not seem ideal. This is a big design decision and should be discussed front and center.

The book has a short and clearly written section on this topic (3.1.10 Substitutes to Certificate Revocation) that we could adapt. This would fit well into the section  "Trust Relations within an Isolation Domain".